### DISCUSSION TITLE
Copy and component updates to 2FA set up pages

### DISCUSSION THREAD

**Discussion by ajacobs21e** [keycloak/keycloak#10637] [Feedback]
The Red Hat downstream version of the mobile authenticator setup pages underwent a copy review by a Content Strategist. These pages were also designed with the latest PatternFly components. We are contributing these changes upstream.

Most changes are text updates and updating to newer PatternFly components (that may not have been available when these pages were originally built).

![2faqr](https://user-images.githubusercontent.com/6415742/157129380-9e1ab40f-1003-4ad9-aaee-8bb22a62c2d0.png)
![2fakey](https://user-images.githubusercontent.com/6415742/157129381-82b37344-9ba8-44f7-890f-802181872fa3.png)

The Red Hat version has download buttons to the Android and Apple app stores (note B) - is this a feature that Keycloak would want as well?
![RH2fa](https://user-images.githubusercontent.com/6415742/157129655-93670807-c116-4f8c-a4c1-e29aa24d3045.png)

https://github.com/keycloak/keycloak/issues/8806

**Comment by mabartos** [comment]
@ajacobs21e Thanks! I really like it! It'd be great to update all components to the PF4 ones. The specific text below the title seems promising as well when the user is properly informed about the required action. However, it'd be great to find out what intention is behind keeping the login theme in the current state; I'm not sufficiently familiar with it.

@ssilvert @stianst Do you have a chance to share your opinion?

**Comment by ssilvert** [comment]
@mabartos LGTM

**Reply by mabartos** [reply]
@ssilvert Thanks!

**Comment by jbman** [comment]
Download buttons in the Keycloak Version would be very helpful, if the app name, the image and download URL is configurable.
This could be done via message key, so that you can define 0-2 the recommended apps by adding realm localization keys.
Maybe the download images could also be omitted having just the app name text and web page configured.

Example:
`mobileAuthenticatorApp1`: FreeOTP
`mobileAuthenticatorLink1`: https://freeotp.github.io/

Would be displayed as:

**1. Install an authenticator app**
[FreeOTP](https://freeotp.github.io/)


Additional note: I would switch one time code and device name input field. Otherwise, the user could think about the device name while the code changes. If device name is typed first, then the user can enter the code and send it immediately.

**Comment by ajacobs21e** [comment]
@jbman Is the device name editable in Keycloak after setting up the authenticator? The one-time code is first since step 3 is to enter this code and it is the required field, whereas device name is optional. Having the required field first is a better hierarchy.

**Reply by jbman** [reply]
@ajacobs21e, the device name is not editable currently. Below you see an example where I named the device 'My mobile'.

The one-time-code changes every 30 seconds while you have the dialog open. If you enter the code first and device name second, the code displayed on your device could have already changed . That's why I recommend to put the one-time-code input field last, even if device name is optional.

![2fa-device](https://user-images.githubusercontent.com/789496/163127178-2dd62995-9a62-4ef7-95a0-36207649e7cc.png)
